code/modules/station_goals/shield.dm

```diff
@@ -0,0 +1,171 @@
+//Station Shield
+// A chain of satellites encircles the station
+// Satellites be actived to generate a shield that will block unorganic matter from passing it.
+/datum/station_goal/station_shield
+	name = "Station Shield"
+
+/datum/station_goal/station_shield/get_report()
+	return {"The station is located in a zone full of space debris.
+			 We have a prototype shielding system you will deploy to reduce collision related accidents.
+
+			 You can order the satellites and control systems through cargo shuttle.
+			 "}
+
+
+/datum/station_goal/station_shield/on_report()
+	//Unlock 
+	var/datum/supply_pack/P = SSshuttle.supply_packs[/datum/supply_pack/misc/shield_sat]
+	P.special_enabled = TRUE
+
+	P = SSshuttle.supply_packs[/datum/supply_pack/misc/shield_sat_control]
+	P.special_enabled = TRUE
+
+/datum/station_goal/station_shield/check_completion()
+	if(..())
+		return TRUE
+	var/list/coverage = list()
+	for(var/obj/machinery/satellite/meteor_shield/A in machines)
+		if(!A.active || A.z != ZLEVEL_STATION)
+			continue
+		coverage |= view(A.kill_range,A)
+	if(coverage.len >= 400)
```

Comment:
This should probably be trackable/specified ingame somehow. Like have the control computer show the current coverage value. There's no ingame way for players to know they need 40 satellites worth of coverage.